### PR TITLE
replaced viper with koanf, feature parity

### LINKED ISSUE #4: remove config reflection
awful code

https://github.com/knadh/koanf might be an (lightweight) alternative to viper

### PR DESCRIPTION
**Pull request by woutslakhorst**
closes #4 

This PR focusses on removing viper and the reflection for config loading. It does not focus on refactoring engines and load/config ordering.

### CODE REVIEW

**Comment by woutslakhorst** [comment]
drop in coverage is due to removing code....

File: core/config.go [hunk 1]
```diff
@@ -21,455 +21,146 @@ package core
...
+	// load into struct
+	if err = ngc.configMap.Unmarshal("", ngc); err != nil {
+		return
 	}
 
 	// initialize logger, verbosity flag needs to be available
-	level, err := log.ParseLevel(ngc.v.GetString(loggerLevelFlag))
-	if err != nil {
-		return err
+	if _, err = log.ParseLevel(ngc.Verbosity); err != nil {
```
Comment (reinkrul) [inline core/config.go:101]: You still have to set the level?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: core/config.go (hunk 1 above)
Comment (woutslakhorst) [reply core/config.go:101]: I think that should be part of #40

File: core/config.go [hunk 2]
```diff
@@ -21,455 +21,146 @@ package core
...
-// LoadConfigFile load the config from the given config file or from the default config file. If the file does not exist it'll continue with default values.
-func (ngc *NutsGlobalConfig) loadConfigFile() error {
-	configFile := ngc.v.GetString(configFileFlag)
-
-	// default path, relative paths and absolute paths should work
-	ngc.v.AddConfigPath(".")
-	ngc.v.SetConfigFile(configFile)
+func (ngc *NutsGlobalConfig) flagSet() *pflag.FlagSet {
+	flagSet := pflag.NewFlagSet("config", pflag.ContinueOnError)
+	flagSet.String(configFileFlag, "", "Nuts config file")
```
[the remark this answers is not in the document]
Comment (woutslakhorst) [reply core/config.go:138]: oversight

File: core/config.go (hunk 1 above)
Comment (reinkrul) [reply core/config.go:102]: Then why do we have the option if it doesn't do anything?

Comment (woutslakhorst) [reply core/config.go:102]: I'll set it here with a todo

**Review by reinkrul — APPROVED** [review]

**Pull request merged**